File: lab/javascript/textos/frequencia_letras.js
function frequenciaLetras(texto) {
    texto = texto.toLowerCase();
    let contagem = {};

    for (let char of texto) {
        if (/[a-zá-úç]/i.test(char)) {  // filtra só letras (aceita acentos simples também)
            contagem[char] = (contagem[char] || 0) + 1;
        }
    }

    let ordenado = Object.keys(contagem).sort();
    for (let letra of ordenado) {
        console.log(`${letra}: ${contagem[letra]}`);
    }
}

frequenciaLetras("Programação em Python!");

File: lab/python/textos/frequencia_letras.py
def frequencia_letras(string):
    string = string.replace(" ", "").lower()
    contagem = {}
    for letra in string:
        if letra.isalpha():
            contagem[letra] = contagem.get(letra, 0) + 1

    string_ordenada = sorted(contagem)
    for letra in string_ordenada:
        print(f"{letra}: {contagem[letra]}")


string = input()
frequencia_letras(string)
